oak_functions/loader/src/server.rs

```diff
@@ -816,15 +831,26 @@ fn oak_functions_resolve_func(field_name: &str) -> Option<(usize, wasmi::Signatu
 /// `wasmi` specific result type `Result<Option<wasmi::RuntimeValue>, wasmi::Trap>`, mapping:
 /// - `Ok(())` to `Ok(Some(OakStatus::Ok))`
 /// - `Err(x)` to `Ok(Some(x))`
-fn map_host_errors(
-    result: Result<(), OakStatus>,
-) -> Result<Option<wasmi::RuntimeValue>, wasmi::Trap> {
+fn oak_to_wasmi(result: Result<(), OakStatus>) -> Result<Option<wasmi::RuntimeValue>, wasmi::Trap> {
     Ok(Some(wasmi::RuntimeValue::I32(result.map_or_else(
         |x: OakStatus| x as i32,
         |_| OakStatus::Ok as i32,
     ))))
 }
 
+/// A helper function to move between our specific result type `Result<(), ChannelStatus>` and the
+/// `wasmi` specific result type `Result<Option<wasmi::RuntimeValue>, wasmi::Trap>`, mapping:
+/// - `Ok(())` to `Ok(Some(ChannelStatus::Ok))`
+/// - `Err(x)` to `Ok(Some(x))`
+fn uwabi_to_wasmi(
+    result: Result<(), ChannelStatus>,
+) -> Result<Option<wasmi::RuntimeValue>, wasmi::Trap> {
+    Ok(Some(wasmi::RuntimeValue::I32(result.map_or_else(
+        |x: ChannelStatus| x as i32,
```

Comment:
Will it cause a problem that the same status (e.g. invalid args) will be represented by different i32 values?

Reply:
I am not sure, I'll investigate and follow up.

Reply:
We will statically know if we expect a ChannelStatus or an OakStatus---so this overlap should be fine!